File: ruby/difference-of-squares/difference_of_squares.rb
VERSION = 1

  class Squares

    attr_reader :numbers

    def initialize(number)
      @numbers = 0.upto(number)
    end

    def square_of_sums
      numbers.reduce(:+) ** 2
    end

    def sum_of_squares
      numbers.map { |number| number ** 2 }.reduce(:+)
    end

    def difference
      square_of_sums - sum_of_squares
    end
  end

File: python/difference-of-squares/difference_of_squares.py
def difference(n):
    return abs(sum_of_squares(n) - square_of_sum(n))

def square_of_sum(n):
    return sum(create_range(n)) ** 2

def sum_of_squares(n):
    squares = map(square, create_range(n))
    return sum(squares)

def square(number):
    return number ** 2

def create_range(n):
    return range(1, n + 1)
